source.go

```diff
@@ -268,10 +276,22 @@ func (s *Source) getPosition(b notion.Block) sdk.Position {
 	)
 }
 
-func (s *Source) getPayload(parent notion.Block, children notion.Blocks) (sdk.RawData, error) {
-	nb := recordPayload{
-		BlockType: parent.GetType().String(),
-		Children:  children,
+func (s *Source) getPayload(ctx context.Context, children notion.Blocks) (sdk.RawData, error) {
+	var payload string
+	for _, c := range children {
+		text, err := extractText(c)
+		if errors.Is(err, errNoExtractor) {
+			sdk.Logger(ctx).Warn().
+				Str("block_type", c.GetType().String()).
+				Msg("no text extractor registered")
+			continue
+		}
+		if err != nil {
+			return nil, err
+		}
+		payload += text
+		payload += "\n"
```

Comment:
nitpick: we can have these two in one line `payload += text + "\n"`

Reply:
noticed that this is repeated a couple of times with attaching a space too, I'm not sure which way looks better (1 line or 2), but I'm fine both ways!